### DISCUSSION TITLE
Record song directly from (selectable) playback device

### DISCUSSION THREAD

**Discussion by mark0100** [marin-m/SongRec#22] [Ideas]
I'm not sure if it's technically possible but it would be very nice if songrec could identify the song i'm listening at on a playback device.
This would be also very handy if you have a single music file which contains multiple songs.

**Comment by marin-m** [comment]
Hello,

I have added a feature allowing to recognize songs from the speakers output in release 0.1.4, which is being pushed to the distribution channels.

It should add a checkbox dubbed "Recognize from my speakers instead of microphone" to the UI.

**Comment by mark0100** [comment]
Great! I'm currently on 0.1.3 on Ubuntu 20.04 but I will definetly check it out when there is a newer version.
Is it not possible to add a dropdown box to select the device to capture from?

**Reply by marin-m** [reply]
The existing dropdown stays. The new checkbox will allow to switch towards the PulseAudio default monitor device (i.e the default system speakers output) as a source. The existing dropdown communicated with ALSA, more exactly with the ALSA compatibility layer of PulseAudio, which did not allow to select a monitor device alone.

**Reply by mark0100** [reply]
Ah I see now. Thank you for your effort making this possible.

**Reply by mark0100** [reply]
I updated SongRec to 0.1.4 and it works fine!
